src/orion/algo/hyperband.py

```diff
@@ -426,10 +485,22 @@ def observe(self, trials):
                 continue
 
     @property
-    def is_done(self):
-        """Return True, if all required execution been done."""
+    def is_done(self) -> bool:
+        """Return True, if all required execution has been done."""
         if self.executed_times >= self.repetitions:
             return True
+        fidelity_index = self.fidelity_index
+        assert fidelity_index is not None
```

Comment:
Another

Reply:
Disagree on this one.
The exception would already have been raised here when instantiating the object:
https://github.com/lebrice/orion/blob/35ce6867769d77028c87f7bb015adfd349315f85/src/orion/algo/hyperband.py#L188-190
Hence using an assert is perfectly adequate here.

Reply:
The alternative would be to overwrite the `fidelity_index` property in this class, and have its annotation be a `str` instead of `str | None`, and raise an error inside the property if there is not fidelity property. Would you like that better?

Reply:
Do you mean doing this for Hyperband (and subclasses) specifically or BaseAlgorithm? That would be reasonable for Hyperband. For BaseAlgorithm we would need a different error message.

Reply:
Fixed in 4680ca57fe0c708393889463c7b326f20b4ee386